src/posix/platform/settings.cpp

```diff
@@ -212,6 +207,10 @@ void otPlatSettingsInit(otInstance *aInstance)
         VerifyOrExit(offset == lseek(sSettingsFd, length, SEEK_CUR), error = OT_ERROR_PARSE);
     }
 
+#if OPENTHREAD_POSIX_CONFIG_SECURE_SETTINGS_ENABLE
```

Comment:
Why don't we have ot::Posix::PlatformSettingsInit and ot::Posix::PlatformSettingsDeinit ?

Reply:
These should not be called by other components, so keeping them private is better.

Reply:
SGTM